lib/styles/components/_buttons.scss

```diff
@@ -3,48 +3,183 @@
 @import '../settings/typography';
+@import '../settings/buttons';
 
-$default-lighter-shadow: 0 4px 12px rgba($color-firefly-black, 0.35);
+$default-lighter-shadow: 0 4px 12px rgba($color-firefly-black-rgb, 0.35);
 
-@mixin outlinedColor($color) {
-	color: rgba($color, 1);
+@mixin setButtonColor($color: '', $color-hovered: '', $color-focused: '', $color-disabled: '') {
+	@if $color != '' {
+		color: $color;
+	}
 
-	&:disabled,
-	&.-disabled {
-		color: rgba($color, 1 - $button-disabled-alpha);
+	@if $color-hovered != '' {
+		&:hover,
+		&.-hovered,
+		&:active,
+		&.-active {
+			color: $color-hovered;
+		}
+	}
+
+	@if $color-focused != '' {
+		&:focus,
+		&.-focused {
+			color: $color-focused;
+		}
+	}
+
+	@if $color-disabled != '' {
+		&.-disabled {
+			color: $color-disabled;
+		}
 	}
 }
 
-@mixin textColor($color, $hover-color: '', $focus-color: '', $disabled-color: '') {
-	@if $hover-color == '' {
-		$hover-color: mix($color, $color-firefly-black, (1 - $button-hover-alpha) * 100%);
+@mixin setButtonBackground(
+	$background: '',
+	$background-hover: '',
+	$background-focus: '',
+	$background-disabled: ''
+) {
+	@if $background != '' {
+		background-color: $background;
+	}
+
+	@if $background == '' {
+		background-color: transparent;
+	}
+
+	@if $background-hover != '' {
+		&:hover,
+		&.-hovered,
+		&:active,
+		&.-active {
+			background-color: $background-hover;
+		}
+	}
+
+	@if $background-focus != '' {
+		&:focus,
+		&.-focused {
+			background-color: $background-focus;
+		}
 	}
 
-	@if $focus-color == '' {
-		$focus-color: mix($color, $color-firefly-black, (1 - $button-focus-alpha) * 100%);
+	@if $background-disabled != '' {
+		&:disabled,
+		&.-disabled {
+			background-color: $background-disabled;
+		}
 	}
+}
 
-	@if $disabled-color == '' {
-		$disabled-color: mix($color, $color-total-white, (1 - $button-disabled-alpha) * 100%);
+@mixin setButtonDisabled(
+	$disabled-color: '',
+	$disabled-icon: '',
+	$disabled-background: '',
+	$disabled-border: ''
+) {
+	@if $disabled-color != '' {
+		&:disabled,
+		&.-disabled {
+			color: $disabled-color;
+		}
 	}
 
-	color: $color;
+	@if $disabled-icon != '' {
+		&:disabled,
+		&.-disabled {
+			.a-button__icon {
+				color: $disabled-icon;
+			}
+		}
+	}
 
-	&:hover,
-	&.-hovered,
-	&:active,
-	&.-active {
-		color: $hover-color;
+	@if $disabled-background != '' {
+		&:disabled,
+		&.-disabled {
+			background-color: $disabled-background;
+		}
 	}
 
-	&:focus,
-	&.-focused {
-		color: $focus-color;
+	@if $disabled-border != '' {
+		&:disabled,
+		&.-disabled {
+			border-color: $disabled-border;
+		}
 	}
+}
 
-	&:disabled,
-	&.-disabled {
-		color: $disabled-color;
+@mixin setButtonAdditions(
+	$ripple: '',
+	$border: '',
+	$icon: '',
+	$icon-hovered: '',
+	$icon-focused: '',
+	$icon-disabled: ''
+) {
+	@if $ripple != '' {
+		.ripple {
+			background-color: $ripple !important;
+		}
+	}
+
+	@if $ripple == '' {
+		.ripple {
+			background-color: transparent !important;
```

Comment:
maybe we could use `display:none` ?

Reply:
Yes, the effect is the same and we get rid of important.